src/server/worker/pipeline/common.go

```diff
@@ -83,7 +83,19 @@ func ReceiveSpout(
 			if err := getNextTarStream(f, out); err != nil {
 				return err
 			}
-			return withSpoutCommit(cancelCtx, pachClient, pipelineInfo, logger, f, func(commit *pfs.Commit, tr *tar.Reader) error {
+			return withSpoutCommit(cancelCtx, pachClient, pipelineInfo, logger, f, func(commit *pfs.Commit, tr *tar.Reader) (retErr error) {
+				putFileClient, err := pachClient.NewPutFileClient()
+				if err != nil {
+					return err
+				}
+				defer func() {
+					if err := putFileClient.Close(); err != nil {
```

Comment:
This is going to overwrite the original error state if an error occurs. We should only set `retErr` if it is nil, which is why I like the pattern where the only check for determining whether to set `retErr` is `retErr == nil` since it is simple and always correct.